src/devices/machine/smc91c9x.cpp

```diff
@@ -394,14 +405,6 @@ int smc91c9x_device::recv_start_cb(u8 *buf, int length)
 		return 0;
 	}
 
-	// discard packets not from WMS
-	if (memcmp(WMS_OUI, &buf[6], 3))
-	{
-		LOGMASKED(LOG_RX, "received non-WMS packet OUI: %02x:%02x:%02x length %d discarded\n", buf[6], buf[7], buf[8], length);
-
-		return 0;
-	}
-
```

Comment:
What was the original intention of this code?  Was it working around some issue elsewhere?

Reply:
smc91c9x was originally written for the Midway/Williams seattle.cpp games, so this code appears to be discarding packets that aren't for those games.  Hopefully Ted or Aaron can verify.

Reply:
I believe that is there because the MAME ethernet bridging causes all lan traffic to be seen by the smc device, and anything that isn't Williams Traffic (non WMS OUI) will cause the lan to fail in the Midway firmware as it isn't expecting any traffic which isn't from another Midway board.

Reply:
Maybe we need to support a filter callback or something so that this can be configurable per driver.

Reply:
That definitely doesn't belong in the MAC device anyway. But with tap/tun on Windows and Linux you could just not bridge your virtual NIC to the host network when that's an issue.

Reply:
Well in order to connect two separate machines you obviously need a physical network to do so, and then you are subject to all of the lan broadcast traffic such as netbios and dhcp.
I don't think the current TCP bitbanger networking is that viable as a solution for Midway linking right now anyways, so the filter could probably be removed for now.
I have a UDP multicast implementation of the bitbanger that I have tested a bit and it seems better because the packets are already filtered due to the unique multicast address, plus you don't need to start the machines in any particular order because UDP is connectionless.

Reply:
Well if you’re doing it between machines, you can do the packet filtering in the firewall between the host network and the virtual tap/tun interface.  I still don’t see an argument for having a hack like this in the device.

Reply:
I don't think anyone's arguing for keeping the hack in the device, and it sounds like Ted's UDP solution would work better for the Williams machines anyway.

If nobody has any further objections to the current state of the PR I'll apply it later tonight (about 8 hours from when I post this).